numpyro/infer/autoguide.py

```diff
@@ -50,13 +52,39 @@ class AutoGuide(ABC):
 
     :param callable model: a pyro model
     :param str prefix: a prefix that will be prefixed to all param internal sites
+    :param callable init_strategy: A per-site initialization function.
+        See :ref:`init_strategy` section for available functions.
+    :param callable create_plates: An optional function inputing the same
+        ``*args,**kwargs`` as ``model()`` and returning a :class:`numpyro.plate`
+        or iterable of plates. Plates not returned will be created
+        automatically as usual. This is useful for data subsampling.
     """
 
-    def __init__(self, model, prefix='auto'):
-        assert isinstance(prefix, str)
+    def __init__(self, model, *, prefix="auto", init_strategy=init_to_uniform, create_plates=None):
         self.model = model
         self.prefix = prefix
+        self.init_strategy = init_strategy
+        self.create_plates = create_plates
         self.prototype_trace = None
+        self._prototype_frames = {}
+        self._prototype_frame_full_sizes = {}
```

Comment:
As mentioned below I believe this is redundant given `self._prototype_frames`

Reply:
In Pyro, each frame is a class (subclass of a namedtuple), so we can set the attribute `full_size` for it. But in NumPyro, it is a namedtuple, which does not allow setting attributes. So I made this workaround.

Reply:
@fritzo You can find descriptions for the differences between NumPyro and Pyro in the first post. :)